### PR TITLE
refactor: add configureProvider plugin action

### PR DESCRIPTION
**Pull request by edvald**
This is part of a larger branch I had, reworking the provider configuration and preparation mechanisms.

### CODE REVIEW

File: garden-service/src/types/plugin/plugin.ts [hunk 1]
```diff
@@ -108,6 +109,21 @@ export interface PluginActionDescription {
...
+  configureProvider: {
+    description: dedent`
+      Validate and transform the given provider configuration.
+
+      Note that this does not need to perform structural schema validation (the framework does that
+      automatically), but should in turn perform semantic validation to make sure the configuration is sane.
+
+      This can also be used to further specify the semantics of the provider, including dependencies.
+
+      This action is called on every execution of Garden, so it should return quickly and avoid doing
```
Comment (thsig) [inline garden-service/src/types/plugin/plugin.ts]: "on every execution of Garden" -> "on every execution of a Garden command"?

Comment (thsig) [inline garden-service/src/types/plugin/plugin.ts]: "avoid doing" -> "avoid"?

File: garden-service/src/types/plugin/plugin.ts [hunk 2]
```diff
@@ -302,6 +321,9 @@ export const moduleActionDescriptions:
       configuration and test configuration. Since services and tests are not specified using built-in
       framework configuration fields, this action needs to specify those via the \`serviceConfigs\` and
       \`testConfigs\` output keys.
+
+      This action is called on every execution of Garden, so it should return quickly and avoid doing
```
Comment (thsig) [inline garden-service/src/types/plugin/plugin.ts]: "avoid doing" -> "avoid"?

File: garden-service/src/types/plugin/plugin.ts [hunk 3]
```diff
@@ -287,6 +303,9 @@ export const moduleActionDescriptions:
       should not specify the built-in fields (such as \`name\`, \`type\` and \`description\`).
 
       Used when auto-generating framework documentation.
+
+      This action is called on every execution of Garden, so it should return quickly and avoid doing
```
Comment (thsig) [inline garden-service/src/types/plugin/plugin.ts]: "avoid doing" -> "avoid"?

File: garden-service/src/types/plugin/plugin.ts (hunk 1 above)
Comment (edvald) [reply garden-service/src/types/plugin/plugin.ts]: How about 
_"Important: This action is called on most executions of Garden commands, so it should return quickly and avoid performing expensive processing or network calls."_

**Pull request merged**